src/mmw/js/src/controllers.js

```diff
@@ -26,6 +27,44 @@ var AppController = {
     }
 };
 
+var ErrorHandlers = {
+    itsi: function() {
+        router.navigate('');
+        alert("We're sorry, but there was an error logging you in with your" +
+            " ITSI credentials. Please log in using another method or" +
+            " continue as a guest."
+        );
+        App.showLoginModal();
+    },
+
+    unknown: function(type) {
+        router.navigate('');
+        console.log("An unknown error occurred: " + type);
+    },
+
+    generic: function() {
+        router.navigate('');
+        alert("We're sorry, but an error occurred in the application.");
+    }
+};
+
+var ErrorController = {
+    error: function(type) {
+        if (type) {
+            switch(type) {
+                case 'itsi':
+                    ErrorHandlers.itsi();
+                    break;
+                default:
+                    ErrorHandlers.unknown(type);
+            }
+        } else {
+            ErrorHandlers.generic();
+        }
+    }
+};
+
 module.exports = {
-    AppController: AppController
+    AppController: AppController,
+    ErrorController: ErrorController
```

Comment:
I was hoping to phase out this top-level controller file (we've been moving the page routes to their respective apps as they've been created). Maybe we could move this to `core/errorController.js`? Curious what others think.